apps/passport-server/src/services/issuanceService.ts

```diff
@@ -423,6 +450,77 @@ export class IssuanceService {
     }
   }
 
+  public async handleCheckInByIdRequest(
+    request: CheckTicketInByIdRequest
+  ): Promise<CheckTicketInByIdResult> {
+    try {
+      const ticketDB = await fetchDevconnectPretixTicketByTicketId(
+        this.context.dbPool,
+        request.ticketId
+      );
+
+      if (!ticketDB) {
+        return {
+          error: { name: "InvalidTicket" },
+          success: false
+        };
+      }
+
+      const ticketData = {
+        ticketId: request.ticketId,
+        eventId: ticketDB?.pretix_events_config_id
+      };
+
+      const checker = await this.checkUserExists(request.checkerProof);
+
+      if (!checker) {
+        return {
+          error: { name: "NotSuperuser" },
+          success: false
+        };
+      }
+
+      const checkerSuperUserPermissions =
+        await fetchDevconnectSuperusersForEmail(
+          this.context.dbPool,
+          checker.email
+        );
+
+      const relevantSuperUserPermission = checkerSuperUserPermissions.find(
+        (perm) => perm.pretix_events_config_id === ticketData.eventId
+      );
+
+      if (!relevantSuperUserPermission) {
+        return { error: { name: "NotSuperuser" }, success: false };
+      }
+
+      const successfullyConsumed = await consumeDevconnectPretixTicket(
+        this.context.dbPool,
+        ticketData.ticketId ?? "",
+        checker.email
+      );
+
+      if (successfullyConsumed) {
+        return {
+          value: undefined,
+          success: true
+        };
+      }
+
+      return {
+        error: { name: "ServerError" },
+        success: false
+      };
+    } catch (e) {
+      logger("Error when consuming devconnect ticket", { error: e });
+      throw new PCDHTTPError(500, "failed to check in", { cause: e });
+    }
+  }
+
+  /**
+   * Check that a ticket is valid for check-in by validating the PCD and
+   * checking the ticket in the DB.
+   */
```

Comment:
worth noting that this is a route that only works for devconnect tickets, not zuzalu tickets or zuconnect tickets.